src/lib/cifs_idmap_sss/cifs_idmap_sss.c

```diff
@@ -304,7 +304,18 @@ int cifs_idmap_ids_to_sids(void *handle, const struct cifs_uxid *cuxid,
     }
 
     for (i = 0; i < num; ++i) {
-        err = sss_nss_getsidbyid((uint32_t)cuxid[i].id.uid, &sid, &id_type);
+        switch (cuxid[i].type) {
+        case CIFS_UXID_TYPE_UID:
+            err = sss_nss_getsidbyuid((uint32_t)cuxid[i].id.uid,
+                                      &sid, &id_type);
+            break;
+        case CIFS_UXID_TYPE_GID:
+            err = sss_nss_getsidbygid((uint32_t)cuxid[i].id.gid,
+                                      &sid, &id_type);
+            break;
+        default:
+            err = sss_nss_getsidbyid((uint32_t)cuxid[i].id.uid, &sid, &id_type);
```

Comment:
I would add a break here as well. I know it's not important because the default statement tends to be last, but I always thought that C got the defaults backwards and that break in a case should be the default :-)

Reply:
I think we typically do not add break to the last (default) case. Especially in this case I wonder if adding the not needed break here would make a reader think if the intention was to break out of the surrounding for-loop because it is not needed to leave the switch statement?